Added area graph and renamed some classes/properties.

lshrinivas opened the pull request:
Got rid of RangeDataSeries and instead folded min/max into numerical
data series. Otherwise, getting aggregate data range to deal with
different types of compatible series was becoming nightmarish

Added some more validation for numeric data series to ensure all
points have the same structure. Also added tests for that

Apply inserts on parent group so action bars can be targetd too.

Review comments + tests for AreaGraph

Fix bug with binned mean plot - need to update
radii on main selection (not just enter selection)
otherwise the order gets messed up.

Added custom legend element for handling complex shapes in the legend

Added test for custom legend labels

Commit:
Added area graph and renamed some classes/properties.

Got rid of RangeDataSeries and instead folded min/max into numerical
data series. Otherwise, getting aggregate data range to deal with
different types of compatible series was becoming nightmarish

Added some more validation for numeric data series to ensure all
points have the same structure. Also added tests for that

Apply inserts on parent group so action bars can be targetd too.

Review comments + tests for AreaGraph

Fix bug with binned mean plot - need to update
radii on main selection (not just enter selection)
otherwise the order gets messed up.

Added custom legend element for handling complex shapes in the legend

Added test for custom legend labels

lshrinivas commented on lib/models/NumericalDataSeries.js:
```diff
@@ -3,19 +3,38 @@ import DataSeries from './DataSeries';
+         *      1. Range for X, Single value for Y - x_min, x_max, y_value (e.g., for a binned mean plot)
+         *      2. Single value for X, Range for Y - x_value, y_min, y_max (e.g., for an area plot)
+         *      3. Single value for X, Range + value for Y - x_value, y_min, y_max, y_value (e.g., for points with error bars)
+         */
```
This is one of the main changes in this PR - I consolidated RangeDataSeries and NumericalDataSeries, since I didn't want an explosion of data series types depending on whether or not one of the axes supported a range. Otherwise we'd need XRangeYSingle, XSingleYRange, XRangeYRange etc. which would be nasty

lshrinivas commented on lib/views/AreaGraph.js:
```diff
@@ -0,0 +1,56 @@
+import Graph from '../views/Graph';
```
This is the other big change - supporting an area graph (basically an envelope of points)

choochburger commented:
You already had PRs up for these changes right? (like [this one](https://github.com/nutonian/nugget/pull/17)) Is there anything new here that we should take a look at? If not, let 'er rip 👍

lshrinivas commented:
Yes - this can only be merged when the feature lands on modular homes though (so hopefully by the end of this sprint). It has been reviewed before and merged and reverted since some of these changes broke existing modular homes code 😄

Commit:
Allow html inside legend labels (for styling)

Commit:
Reduce space wastage on x axis label

Redraw guide layer after axes so guide elements don't show up behind the axis labels

Remove override on YY comparison guide

Allow overrides for scattergraph guide colors

lshrinivas committed:
Merge pull request #24 from nutonian/p_axis_labels

Update styles of guides [re #6226, #6217]

Commit:
Apply inserts on scatter graph after drawing error bars
(otherwise inserts on error bars have stale data)

lshrinivas committed:
Merge pull request #25 from nutonian/p_uat_feedback

Apply inserts on scatter graph after drawing error bars

choochburger committed:
add method to update axis labels

choochburger committed:
Merge pull request #26 from nutonian/cjl_6944_update_axis_labels

Add method to update axis labels [re #6944]
